packages/core/test/session.test.ts

```diff
@@ -474,6 +476,78 @@ describe('session', () => {
         expect.objectContaining({ auth })
       )
     })
+
+    it('should round up sub milliseconds transaction timeouts', async () => {
+      return await fc.assert(
+        fc.asyncProperty(
+          fc.float({ min: 0, noNaN: true }),
+          async (timeout: number) => {
+            const connection = mockBeginWithSuccess(newFakeConnection())
+
+            const { session } = setupSession({
+              connection,
+              beginTx: false,
+              database: 'neo4j'
+            })
+
+            await session.beginTransaction({ timeout })
+
+            expect(connection.seenBeginTransaction[0][0].txConfig.timeout)
+              .toEqual(int(Math.ceil(timeout)))
+          }
+        )
+      )
+    })
+
+    it('should log a warning for timeout configurations with sub milliseconds', async () => {
```

Comment:
```suggestion
    it('should log when a timeout is configured with sub milliseconds', async () => {
```
it's now a warning... well sort of, but not on warning level, either way ;)